compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1470,7 +1470,7 @@ pub struct Statement<'tcx> {
 
 // `Statement` is used a lot. Make sure it doesn't unintentionally get bigger.
 #[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
-static_assert_size!(Statement<'_>, 32);
+static_assert_size!(Statement<'_>, 40);
```

Comment:
Hmm, this might be a bit of an issue

Reply:
We're making all statements bigger here which seems non-ideal.

Reply:
We could refactor `FakeRead` to have a `Box<FakeReadData>` which contains the `FakeReadCause` and the `Place<'tcx>` (no need to box it separately)

Reply:
I added a `box (FakeReadCause, Place<'tcx>)`